dlib/image_transforms/draw.h

```diff
@@ -423,44 +424,53 @@ namespace dlib
 // ---------------------------------------------------------------------------------------
 
     template <typename image_type, typename pixel_type>
-    void fill_convex_polygon (
+    void draw_solid_convex_polygon (
         image_type& image,
         const polygon& poly,
-        const pixel_type& pixel,
+        const pixel_type& color,
         const bool antialias = true,
         const rectangle& area = rectangle(std::numeric_limits<long>::min(), std::numeric_limits<long>::min(),
                                           std::numeric_limits<long>::max(), std::numeric_limits<long>::max())
     )
     {
-        using std::max;
-        using std::min;
         const rectangle valid_area(get_rect(image).intersect(area));
-
         const rectangle bounding_box = poly.get_rect();
 
         // Don't do anything if the polygon is totally outside the area we can draw in
         // right now.
         if (bounding_box.intersect(valid_area).is_empty())
             return;
 
+        image_view<image_type> img(image);
         rgb_alpha_pixel alpha_pixel;
-        assign_pixel(alpha_pixel, pixel);
+        assign_pixel(alpha_pixel, color);
         const unsigned char max_alpha = alpha_pixel.alpha;
 
         // we will only want to loop over the part of left_boundary that is part of the
         // valid_area.
-        long top = max(valid_area.top(),bounding_box.top());
-        long bottom = min(valid_area.bottom(),bounding_box.bottom());
+        long left = std::max(valid_area.left(), bounding_box.left());
+        long top = std::max(valid_area.top(), bounding_box.top());
+        long right = std::min(valid_area.right(), bounding_box.right());
+        long bottom = std::min(valid_area.bottom(), bounding_box.bottom());
```

Comment:
This could be simplified a little.  This is computing `valid_area.intersect(bounding_box)`.  So you could replace `valid_area` with that new smaller box and then simplify some of this a bit.  Like if `antialias` was true you could do `grow_rect(valid_area.intersect(bounding_box)),1)` instead of some of the logic below, assuming I'm reading it right.